integrations/octopus/client.py

```diff
@@ -0,0 +1,115 @@
+from typing import Any, AsyncGenerator, Optional
+from loguru import logger
+from port_ocean.utils import http_async_client
+from httpx import HTTPStatusError, Timeout
+
+PAGE_SIZE = 50
+
+
+class OctopusClient:
+    def __init__(self, server_url: str, octopus_api_key: str) -> None:
+        self.octopus_url = f"{server_url.rstrip('/')}/api/"
+        self.api_auth_header = {"X-Octopus-ApiKey": octopus_api_key}
+        self.client = http_async_client
+        self.client.timeout = Timeout(60)
+        self.client.headers.update(self.api_auth_header)
+
+    async def _send_api_request(
+        self,
+        endpoint: str,
+        params: Optional[dict[str, Any]] = None,
+        json_data: Optional[dict[str, Any]] = None,
+        method: str = "GET",
+    ) -> Any:
+        """Send a request to the Octopus Deploy API."""
+        url = f"{self.octopus_url}{endpoint}"
+        response = await self.client.request(
+            url=url,
+            method=method,
+            headers=self.api_auth_header,
+            params=params,
+            json=json_data,
+        )
+        try:
+            response.raise_for_status()
+        except HTTPStatusError as e:
+            logger.error(
+                f"Got HTTP error to url: {url} with status code: {e.response.status_code} and response text: {e.response.text}"
+            )
+            raise
+        return response.json()
+
+    async def get_paginated_resources(
+        self,
+        kind: str,
+        params: Optional[dict[str, Any]] = None,
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        """Fetch paginated data from the Octopus Deploy API."""
+        if params is None:
+            params = {}
+        params["skip"] = 0
+        params["take"] = PAGE_SIZE
+        page = 0
+        while True:
+            response = await self._send_api_request(f"{kind}s", params=params)
+            items = response.get("Items", [])
+            last_page = response.get("LastPageNumber", 0)
+            yield items
+            if page >= last_page:
+                break
+            params["skip"] += PAGE_SIZE
+            page += 1
+
+    async def get_single_entity(
```

Comment:
Is this item called "entity" in Octopus as well? If not, lets change it
The "entity" terminology should be saved to Port related entities

Reply:
I have changed this to resource